test/node-rpc-test.js

```diff
@@ -119,4 +221,296 @@ describe('RPC', function() {
+      // create wallet for testing.
+      await wclient.createWallet(TEST_WALLET);
+      await wclient.execute('selectwallet', [TEST_WALLET]);
+      const testAddr = await wclient.execute('getnewaddress');
+
+      // back to miner
+      await wclient.execute('selectwallet', [MINER_WALLET]);
+      const minerAddr = await wclient.execute('getnewaddress');
+
+      { // block 0 - This block will have 2 txs (including coinbase)
+        const txid = await wclient.execute('sendtoaddress', [testAddr, 1]);
+        const [bh] = await nclient.execute('generatetoaddress', [1, minerAddr]);
+
+        // this txid will be spent in block 2.
+        txids.push([txid]);
+        blockhashes.push(bh);
+      }
+
+      { // block 1 - Spend utxo from block 1 in this block. (2txs)
+        await wclient.execute('selectwallet', [TEST_WALLET]);
+
+        const txid = await wclient.execute('sendtoaddress', [minerAddr, 1, '', '', true]);
+        const [bh] = await nclient.execute('generatetoaddress', [1, minerAddr]);
+
+        txids.push([txid]);
+        blockhashes.push(bh);
+
+        const unspent = await wclient.execute('listunspent');
+        assert.strictEqual(unspent.length, 0);
+
+        await wclient.execute('selectwallet', [MINER_WALLET]);
+      }
+
+      { // block 2 - This block will have 4 txs (including coinbase)
+        const blocktxs = [];
+        for (let i = 0; i < 3; i++) {
+          const txid = await wclient.execute('sendtoaddress', [testAddr, 1]);
+          blocktxs.push(txid);
+        }
+
+        const [bh] = await nclient.execute('generatetoaddress', [1, minerAddr]);
+
+        txids.push(blocktxs);
+        blockhashes.push(bh);
+      }
+
+      { // block 3 - This block will have 7 txs (including coinbase)
+        const blocktxs = [];
+        for (let i = 0; i < 6; i++) {
+          const txid = await wclient.execute('sendtoaddress', [testAddr, 1]);
+          blocktxs.push(txid);
+        }
+
+        const [bh] = await nclient.execute('generatetoaddress', [1, minerAddr]);
+
+        txids.push(blocktxs);
+        blockhashes.push(bh);
+      }
+
+      { // mempool tx
+        mempoolTXID = await wclient.execute('sendtoaddress', [testAddr, 1]);
+        await forValue(fullnode.mempool.map, 'size', 1);
+        await forValue(fullnode.mempool.map, 'size', 1);
+      }
+
+      const bh = blockhashes[blockhashes.length - 1];
+      const {height} = await nclient.execute('getblock', [bh]);
+
+      await forValue(fullnode.chain, 'height', height);
+      await forValue(fullindexed.txindex, 'height', height);
+      await forValue(prunednode.chain, 'height', height);
+      await forValue(spvnode.chain, 'height', height);
+    });
+
+    it('should fail in spv mode', async () => {
+      await assert.rejects(async () => {
+        await spvnclient.execute('gettxoutproof', [[mempoolTXID]]);
+      }, {
+        name: 'Error',
+        type: 'RPCError',
+        message: /SPV/,
+        code: -1 >>> 0
```

Comment:
This trick here, help with the bug in bcurl: https://github.com/bcoin-org/bcurl/pull/16